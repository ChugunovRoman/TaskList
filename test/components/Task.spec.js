'use strict';

import React from 'react';
import sinon from 'sinon';

import Task from '../../src/components/Task';

describe('Components: <Task />', () => {

    it('should parse props with mount', () => {
        const value = '0';
        const onComplete = sinon.spy();

        const props = {
            key: "0",
            id: "0",
            name: "Name",
            description: "description of task",
            onComplete: onComplete
        };

        const wrapper = mount (
            <Task {...props} />
        );

        expect(wrapper.prop('id')).toEqual('0');
        expect(wrapper.prop('name')).toEqual('Name');
        expect(wrapper.find('p').at(0).text()).toEqual('Name');
        expect(wrapper.prop('description')).toEqual('description of task');
        expect(wrapper.find('p').at(1).text()).toEqual('description of task');
        expect(wrapper.find('div')).toHaveLength(3);

        expect(wrapper).toMatchSnapshot();

        wrapper.find('button.desc__btn').simulate('click', { target: { value } });
        expect(onComplete.calledOnce).toEqual(true);

        expect(wrapper).toMatchSnapshot();
    });
});
