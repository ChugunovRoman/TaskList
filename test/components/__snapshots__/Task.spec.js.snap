// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: <Task /> should parse props with mount 1`] = `
<Task
  description="description of task"
  id="0"
  name="Name"
  onComplete={[Function]}
>
  <div
    className="task"
  >
    <div
      className="task__name"
      onClick={[Function]}
    >
      <p>
        Name
      </p>
    </div>
    <div
      className="task__desc desc"
    >
      <p
        className="desc__text"
      >
        description of task
      </p>
      <button
        className="desc__btn btn"
        disabled=""
        onClick={[Function]}
        value="0"
      >
        Завершить
      </button>
    </div>
  </div>
</Task>
`;

exports[`Components: <Task /> should parse props with mount 2`] = `
<Task
  description="description of task"
  id="0"
  name="Name"
  onComplete={[Function]}
>
  <div
    className="task"
  >
    <div
      className="task__name"
      onClick={[Function]}
    >
      <p>
        Name
      </p>
    </div>
    <div
      className="task__desc desc"
    >
      <p
        className="desc__text"
      >
        description of task
      </p>
      <button
        className="desc__btn btn"
        disabled=""
        onClick={[Function]}
        value="0"
      >
        Завершить
      </button>
    </div>
  </div>
</Task>
`;
